File: app/src/main/java/ie/dublinbuspal/android/data/local/Converters.java
package ie.dublinbuspal.android.data.local;

import androidx.room.TypeConverter;
import ie.dublinbuspal.android.util.CollectionUtilities;
import ie.dublinbuspal.android.util.StringUtilities;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public final class Converters {

    private final static String DELIMETER = "::";

    private Converters() {
        throw new UnsupportedOperationException();
    }

    @TypeConverter
    public static String fromList(List<String> list) {
        if (CollectionUtilities.isNullOrEmpty(list)) {
            return null;
        }
        return StringUtilities.join(list, DELIMETER);
    }

    @TypeConverter
    public static List<String> toList(String string) {
        if (string != null) {
            return Arrays.asList(string.split(DELIMETER));
        }
        return Collections.emptyList();
    }

}

File: data/src/main/kotlin/ie/dublinbuspal/data/Converters.kt
package ie.dublinbuspal.data

import androidx.room.TypeConverter
import ie.dublinbuspal.util.CollectionUtils
import ie.dublinbuspal.util.StringUtils
import org.threeten.bp.Instant
import java.util.*

object Converters {

    private const val DELIMETER = "::"

    @TypeConverter
    @JvmStatic
    fun fromList(list: List<String>): String {
        if (CollectionUtils.isNullOrEmpty(list)) {
            return DELIMETER
        }
        return StringUtils.join(list, DELIMETER)
    }

    @TypeConverter
    @JvmStatic
    fun toList(value: String): List<String> {
        if (value == DELIMETER) {
            return Collections.emptyList()
        }
        return value.split(DELIMETER)
    }

    @TypeConverter
    @JvmStatic
    fun fromInstant(instant: Instant): Long {
        return instant.epochSecond
    }

    @TypeConverter
    @JvmStatic
    fun toInstant(value: Long): Instant {
        return Instant.ofEpochSecond(value)
    }

}
